apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -65,10 +69,131 @@ export const loadTransactionResponseMetadata = createAsyncThunk<
+    receiverAddress: string;
+};
+
+export type TxnMetaResponse = {
+    objectIDs: string[];
+    coins: CoinsMetaProps[];
+};
+
+const getEventsSummary = (
+    txEffects: TransactionEffects,
+    address: string
+): TxnMetaResponse => {
+    const events = txEffects?.events || [];
+
+    const coinsMeta = events
+        .map((event) => {
+            if (
+                !('coinBalanceChange' in event) ||
+                (event?.coinBalanceChange?.changeType !== 'Receive' &&
+                    event?.coinBalanceChange?.changeType !== 'Pay')
+            ) {
+                return null;
+            }
+            /// Combine all the coin balance changes from Pay and Receive events
+            /// A net positive amount means the user received coins
+            /// A net negative amount means the user sent coins
+            const { coinBalanceChange } = event;
+            const { coinType, amount, coinObjectId, owner } = coinBalanceChange;
+            const { AddressOwner } = owner as { AddressOwner: string };
+            const { ObjectOwner } = owner as { ObjectOwner: string };
+
+            if (ObjectOwner) {
+                return null;
+            }
+
+            return {
+                amount: amount,
+                coinType: coinType,
+                coinObjectId: coinObjectId,
+                receiverAddress: AddressOwner,
+            };
+        })
+        .filter(notEmpty);
+
+    const objectIDs: string[] = events
+
+        .map((event) => {
+            if (!('transferObject' in event)) {
+                return null;
+            }
+            const { transferObject } = event;
+            const { AddressOwner } = transferObject.recipient as {
+                AddressOwner: string;
+            };
+            if (AddressOwner !== address) {
+                return null;
+            }
+            return transferObject?.objectId;
+        })
+        .filter(notEmpty);
+
+    /// Group coins by receiverAddress
+    // sum coins by coinType for each receiverAddress
+    const meta = coinsMeta.reduce((acc, value, _) => {
+        return {
+            ...acc,
+            [`${value.receiverAddress}${value.coinType}`]: {
+                amount:
+                    value.amount +
+                    (acc[`${value.receiverAddress}${value.coinType}`]?.amount ||
+                        0),
+                coinType: value.coinType,
+                receiverAddress: value.receiverAddress,
+            },
+        };
+    }, {} as { [coinType: string]: CoinsMetaProps });
+
+    return {
+        objectIDs,
+        coins: Object.values(meta),
+    };
+};
+
+export const executeDryRunTransactionRequest = createAsyncThunk<
+    {
+        txRequestID: string;
+        txnMeta?: TxnMetaResponse;
+        txGasEstimation?: number;
+    },
+    {
+        txData: string | SignableTransaction | Base64DataBuffer;
+        id: string;
+    },
+    AppThunkConfig
+>(
+    'dryrun-transaction-request',
```

Comment:
We should prioritize moving this out of redux and into state using `useQuery`. We should try to stop adding more state to Redux. If you want to just get this PR out we can land this as-is but I really want us to stop this ASAP.